backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/Migrations.java

```diff
@@ -127,12 +128,26 @@ public static synchronized void initOrMigrate(SqlDatabase db) {
             executeMigrationFile(
                 tdb, "migration21.sql", "add exist and range role to schemas and metadata");
 
-          if (version < 22)
+          if (version < 22) {
             executeMigrationFile(
                 tdb,
                 "migration22.sql",
                 "remove physical refback triggerfunctions (cascade to include triggers) and then the columns");
 
+            // doing this in java here might make next migrations impossible
+            // however the migration is impossible in sql because of the complex composite key name
+            // logic
```

Comment:
Why would this make future migrations impossible?

Reply:
not impossible but more difficult because in principle you need to do first the sql changes before running the java bits. (we used to do migrations in java only but then the current version of emx2 couldn't work with a previous version of the schema and the migration broke; however this migration is pretty innocent so can also be done as last step in next version)